### ISSUE TITLE
qemu-guest-agent breaks Calico networking

### ISSUE THREAD

**Issue by stoyanr** [gardener/gardener-extension-provider-kubevirt#33]
We would like to install and start the `qemu-guest-agent` on Kubevirt VMs to enable propagating the IP addresses we get for Multus network interfaces to the VM and from there, via the CCM, to the Kubernetes node, as this is important for debugging and for other reasons. We even proposed a PR for this for our Ubuntu OS extension, see https://github.com/gardener/gardener-extension-os-ubuntu/pull/21.

Unfortunately, it turned out that adding `qemu-guest-agent` leads to the issue that `calico-node` pods in the shoot cluster are no longer able to start with errors similar to:

```
Liveness probe failed: calico/node is not ready: bird/confd is not live: Service bird is not running. Output << down: /etc/service/enabled/bird: 0s, normally up, want up >>
```

As a result, the shoot network is not properly initialized, the Gardener reconciliation fails, and the Shoot is not usable. The issue is present independently if there are any Multus network attached, with just the default pod network the issue is still there.

We suspect that the issue is related to the Calico IP autodetection mechanism, see https://docs.projectcalico.org/networking/ip-autodetection. We tried workaround such as setting the autodetection method to `interface=enp1s0` and setting the IP explicitly, but none worked.

Another issue that we see with `qemu-guest-agent` is that it propagates the IP together with the mask to the VM, e.g. 10.225.128.106/32 instead of 10.225.128.106. This is propagated also the Kubernetes node by the CCM and leads to other issues, such as being unable to stream logs from the VM.

We suspect this is a Kubevirt bug and we should try to fix it upstream.

/area networking
/kind bug
/priority normal
/platform kubevirt

**Comment by mfranczy** [comment]
I found the problem, I am going to work on a KubeVirt patch.

**Comment by mfranczy** [comment]
The issue is only related to IP address format, that's why calico cannot start
```
Error from server: failed to construct request for https://10.x.x.106%2F32:10250/containerLogs/kube-system/calico-node-lvcgj/calico-node, got parse https://10.225.128.106%2F32:10250/containerLogs/kube-system/calico-node-lvcgj/calico-node: invalid URL escape "%2F"
```

**Comment by mfranczy** [comment]
I was able to fix ip reporting on the KubeVirt side, still didn't created an upstream PR, first I have to confirm that everything works fine.

**Comment by mfranczy** [comment]
@stoyanr we found a common ground in the qemu-guest-agent PR so I have updated it, I will close the issue once it is merged.

**Comment by mfranczy** [comment]
the PR is merged.
